Getting started card on Add page and Cluster Overview

gdoyle1 opened the pull request:
This PR adds design documentation for the Getting Started card that will be added on the Add page and Cluster Overview **replacing** the existing quick start card in 4.8.

PTAL
@openshift/team-ux-leads
@openshift/team-ux-review (Administrator perspective)
@openshift/team-devconsole-ux (Developer perspective)
@openshift/team-kni-ux (KNI & Virtualization)

andybraren reviewed (commented):
Thank you Gina! Cool to see this becoming real. 🙂

Just a couple of small nits and a question about the label behavior from me.

gdoyle1 commented on designs/administrator/4.8/getting-started-card/index.md:
```diff
@@ -0,0 +1,67 @@
+
+Selecting the label will bring the card back onto the dashboard.
+
+![add-page-label-hover](img/add-page-label-hover.png)
```
No there wouldn't be :/ It would be fully dismissing the "hint"

gdoyle1 commented:
Updated @andybraren ! I also had to update some of the language being used after Abi reviewed

itsptk commented on designs/administrator/4.8/getting-started-card/index.md:
```diff
@@ -0,0 +1,67 @@
+## Card design and interactions
+The design of the card will remain consistent on the Admin and Developer perspectives, but the content within each column will vary. The reason for this is because there is featured content we would like users to see depending on the perspective they land on.
+
+![card](img/card-design.png)
```
I guess I'd mention that because this image calls out these are on 'connected clusters' I sort of expected an equivalent image or guidance on what to do if the cluster is disconnected, which doesn't seem to be uncommon. Can there be less than 3 columns?

gdoyle1 replied:
@itsptk We decided the same content would be shown if you have a connected or disconnected cluster, so I might just cut out the pink title there.

Yeah there can be less than 3 columns depending on content you are dealing with, if validation is included (which then a column would disappear) - I should probably add that in there

itsptk commented on designs/administrator/4.8/getting-started-card/index.md:
```diff
@@ -0,0 +1,67 @@
+![add-page-dismiss](img/add-page-dismiss.png)
+![cluster-overview-hidden](img/cluster-overview-hidden.png)
+
+Once the card is hidden, a label will appear in the page header, allowing the user to bring it back. This label acts more as a "hint" and is dismissible itself, in case the user has zero interest in seeing anything getting started related.
```
I wonder if some sort of like link or expandable control could be used in place of a label here? The use here sort of feels like more prominent than I'd expect for something the user just expressed interest in hiding. Perhaps if it was more discrete we would also not need a way to remove that control entirely, maybe addressing always giving a way to add this card back.

Also selfishly I know we are using a similar label to highlight quick starts on pages they relate to so I don't want the user to start to ignore them. 😁

beaumorley replied:
+1 to Peter's comment. At least on the developer side, I think we need to come up with a more holistic solution for showing and hiding. I'd like to see the "show getting started label" and the "Details on" toggle combined.

On the Topology view, we have a display dropdown that lets the user configure their display. It has different controls within the dropdown. Might be a good area to get some user feedback post 4.8 on different options...

gdoyle1 replied:
@itsptk It's too late to change that for this first pass since it's already been approved by stakeholders BUT this issue should be solved in future iterations by one of the following:
- User can access Getting started resources in the help menu , where that catalog page would replace the existing Quick starts page. That way you can access the information on the page rather than bringing the card back
- Use some sort of toggle (like @beaumorley was saying) in page heading to allow for the card to be brought back (we would combine this with the details on/off switch on the developer side)
- When customizable dashboards are introduced, which both Admin and Dev PMs have already been discussing, we can include the ability to hide/show the card and move it around with that functionality

Hopefully that helped! I can add those ideas in the future section as well

itsptk reviewed (commented):
Added a few random thoughts/comments, looks good! Further guidance for the user will surely be appreciated.

gdoyle1 commented:
@beaumorley For this comment https://github.com/openshift/openshift-origin-design/pull/520#discussion_r608084589 , @rachael-phillips and I are working with SJ to get that tested at Summit!

gdoyle1 commented:
@beaumorley @itsptk made some more updates based on the conversations you brought up! I added more detail in future considerations for the points you brought up. And then for guidance on how to reuse the card component, I added a new section

beanh66 reviewed (approved):
LGTM
